### PR TITLE
[lua] Pet magic burst message and accbonus

### PR DESCRIPTION
**Pull request by MowFord**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Implements "Magic Burst!" messages for smn petskills

The `JA_MAGIC_BURST` message is only valid for petskill-type actions, so as the rest of the smn abilities get moved over they will all get the message.

example of MB on the primary target
image

example of MB on secondary target
image

Nether blast message unchanged as it doesn't use the petskill action type yet (Note that it still gets bonus acc/damage)
image

Confirmed wyvern breaths already have the MB message, so the final thing would be jug pets for bst. None of these are moved to petskills as far as I know, but as they do this logic will catch them due to applying when `petID() >0` and `actionType== xi.action.PET_MOBABILITY_FINISH`

Bst Jug pets currently use mobskill lua files, so they (correctly) do not get the call to `xi.mobskills.mobAddBonuses`, but when they get reworked their lua should mimic avatar blood pact lua files and it should "just work"

Confirmation that bst jugs should magic burst from [this retail update](https://forum.square-enix.com/ffxi/threads/46068-Feb-19-2015-(JST)-Version-Update)
image

Note: this also fixes the default damage message for petskills, as they should be using `xi.msg.basic.USES_JA_TAKE_DAMAGE` instead of the mobskill/weaponskill `xi.msg.basic.DAMAGE` message id.

## Steps to test these changes

create skillchains and use various abilities to see you get magic burst if the elements match.

### CODE REVIEW

File: scripts/globals/mobskills.lua [hunk 1]
```diff
@@ -375,6 +383,15 @@ xi.mobskills.mobAddBonuses = function(caster, target, dmg, ele) -- used for SMN
     dmg             = math.floor(dmg * dayWeatherBonus)
 
     local burst = calculateMobMagicBurst(caster, ele, target)
+    if
+        burst > 1.0 and
+        caster:getPetID() >= 0 -- all pets except charmed pets can get magic burst message, but only with petskill action
```
Comment (WinterSolstice8) [inline scripts/globals/mobskills.lua]: `>=`?

Comment (MowFord) [reply scripts/globals/mobskills.lua]: Multiplier is default 1

Oh you mean pet id, whoops

Comment (MowFord) [reply scripts/globals/mobskills.lua]: Fixed, nice catch

File: scripts/globals/mobskills.lua [hunk 2]
```diff
@@ -288,15 +288,23 @@ xi.mobskills.mobMagicalMove = function(mob, target, skill, damage, element, dmgm
...
-            avatarAccBonus = utils.clamp(master:getSkillLevel(xi.skill.SUMMONING_MAGIC) - master:getMaxSkillLevel(mob:getMainLvl(), xi.job.SMN, xi.skill.SUMMONING_MAGIC), 0, 200)
+            petAccBonus = utils.clamp(master:getSkillLevel(xi.skill.SUMMONING_MAGIC) - master:getMaxSkillLevel(mob:getMainLvl(), xi.job.SMN, xi.skill.SUMMONING_MAGIC), 0, 200)
+        end
+
+        local skillchainTier, _ = xi.magicburst.formMagicBurst(element, target)
+        if
+            mob:getPetID() > 0 and
+            skillchainTier > 0
+        then
+            petAccBonus = petAccBonus + 25
```
Comment (WinterSolstice8) [inline scripts/globals/mobskills.lua:303]: Where does this 25 bonus macc come from? Any source?

File: scripts/globals/mobskills.lua [hunk 3]
```diff
@@ -375,6 +383,15 @@ xi.mobskills.mobAddBonuses = function(caster, target, dmg, ele) -- used for SMN
     dmg             = math.floor(dmg * dayWeatherBonus)
 
     local burst = calculateMobMagicBurst(caster, ele, target)
+    if
+        burst > 1.0 and
+        caster:getPetID() > 0 -- all pets except charmed pets can get magic burst message, but only with petskill action
+    then
+        target:setLocalVar('abilityMagicBurst', 1)
```
Comment (WinterSolstice8) [inline scripts/globals/mobskills.lua]: Not a fan of using local vars unless absolutely necessary. the two kinds of `skill` we feed in could have something like... `isMagicBurst` and `setMagicBurst`, what do you think?

File: scripts/globals/mobskills.lua (hunk 2 above)
Comment (MowFord) [reply scripts/globals/mobskills.lua:303]: Pulled this from an xi.magic function. I assumed it was a global bonus during a magic burst

Comment (WinterSolstice8) [reply scripts/globals/mobskills.lua:303]: yeah Jimmy seems to think it should be 30? he's looking into it so standby

File: scripts/globals/mobskills.lua (hunk 3 above)
Comment (MowFord) [reply scripts/globals/mobskills.lua]: This function does not pass in the skill. If we pass the skill we could set the message directly.

I assumed deciding the message in a single place (final adjustments function) was better

Comment (WinterSolstice8) [reply scripts/globals/mobskills.lua]: I agree that setting the message in final adjustments makes sense, but I think passing in `skill` here and setting the flag to be read later would be ideal

Comment (MowFord) [reply scripts/globals/mobskills.lua]: I have an idea of what to do to keep the simplicity this PR currently has. Will work on it later today

File: scripts/globals/mobskills.lua (hunk 2 above)
Comment (MowFord) [reply scripts/globals/mobskills.lua:303]: I've updated to +30, lmk if it ends up being something else

File: scripts/globals/mobskills.lua (hunk 3 above)
Comment (MowFord) [reply scripts/globals/mobskills.lua]: Updated all petskill lua to pass the respective skill object and the magic burst message is set if the skill is there

then in the "final adjustment" functions, I check the message only if it's a pet_finish action.

Spot checked that it still works as expected

Comment (MowFord) [reply scripts/globals/mobskills.lua]: Touching all these files made me notice that Ifrit's abilities are getting bonus dmg from skillchains:

image

I assume this isn't correct

**Comment by Xaver-DaRed** [comment]
Needs rebase

File: scripts/globals/mobskills.lua (hunk 2 above)
Comment (WinterSolstice8) [reply scripts/globals/mobskills.lua:303]: Looks like it's 25, there's some ancient blog that says that

File: scripts/globals/mobskills.lua (hunk 3 above)
Comment (WinterSolstice8) [reply scripts/globals/mobskills.lua]: I don't recall this being a thing, yeah

Comment (WinterSolstice8) [reply scripts/globals/mobskills.lua]: I think we can leave that alone for now and fix it later unless it's egregious levels of extra damage?

File: scripts/globals/summon.lua [hunk 4]
```diff
@@ -278,7 +278,15 @@ xi.summon.avatarFinalAdjustments = function(dmg, mob, skill, target, skilltype,
 
     -- set message to damage
     -- this is for AoE because its only set once
-    skill:setMsg(xi.msg.basic.DAMAGE)
+    if
+        mob:getCurrentAction() == xi.action.PET_MOBABILITY_FINISH
+    then
```
Comment (Xaver-DaRed) [inline scripts/globals/summon.lua]: Theres only one condition here. It can be one line

File: scripts/globals/mobskills.lua [hunk 5]
```diff
@@ -485,7 +501,15 @@ xi.mobskills.mobFinalAdjustments = function(dmg, mob, skill, target, attackType,
 
     -- set message to damage
     -- this is for AoE because its only set once
-    skill:setMsg(xi.msg.basic.DAMAGE)
+    if
+        mob:getCurrentAction() == xi.action.PET_MOBABILITY_FINISH
+    then
```
Comment (Xaver-DaRed) [inline scripts/globals/mobskills.lua]: This is also only one condition. Could be just one line.

**Review by Xaver-DaRed — APPROVED** [review]
Approving anyway. This file needs a cleanup anyway.

File: scripts/globals/mobskills.lua (hunk 2 above)
Comment (MowFord) [reply scripts/globals/mobskills.lua:303]: snip snap. Updated back to +25. Thanks for verifying

File: scripts/globals/summon.lua (hunk 4 above)
Comment (MowFord) [reply scripts/globals/summon.lua]: Done

File: scripts/globals/mobskills.lua (hunk 5 above)
Comment (MowFord) [reply scripts/globals/mobskills.lua]: Done

**Review by WinterSolstice8 — APPROVED** [review]

**Pull request merged**